### DISCUSSION TITLE
How to get whole list of components in simulated results?

### DISCUSSION THREAD

**Discussion by Uzair1shah** [BioSTEAMDevelopmentGroup/biosteam#165] [Q&A] [answered]
Hello,

I am working with a feedstock which has 9 to 10 components. When I try to simulate any unit operation, it truncates and combines the last two components and represents it as ". . .". It makes it difficult to check the mass balance of each component in various unit operations. The snapshot of the problem is attached below:
![image](https://github.com/BioSTEAMDevelopmentGroup/biosteam/assets/128337604/7ec55d23-59e2-4493-9f50-4a4e5a374eba)

And the code for simulation is as follows:
```
# Rotary Dryer
S107 = Stream('S107', phase='g') #Air
S108 = Stream('S108', phase='g') #Natural gas
S109 = Stream('S109', phase='s') #dry feedstock/solids
S110 = Stream('S110', phase='g') #hot air
S111 = Stream('S111', phase='g') #emissions
UO105 = units.DrumDryer('UO105', ins=(S105,S107,S108), outs=(S109,S110,S111), split = 0.5 )

# Simulation of UO105 (Rotary dryer)
UO105.simulate()
UO105.show(flow='kg/hr')
r5 = UO105.results()
UO105.diagram()
print(r5)
```
I wanted to ask if there is a way that the simulated results do not truncate the last two/three components and combine them. If not, is there a way to check the flow of each component as we have defined each component already in the code. Thanks a lot.

**Comment by yoelcortes** [comment, marked as answer]
@Uzair1shah,

This chapter of the documentation shows how it's done:
https://biosteam.readthedocs.io/en/latest/tutorial/Preferences.html#Stream-data-display-preferences

Thanks!

**Reply by yoelcortes** [reply]
Also, here the documentation for the show method:
https://biosteam.readthedocs.io/en/latest/API/thermosteam/Stream.html#thermosteam.Stream.show
